p2p/ipld/plugin/plugin.go

```diff
@@ -0,0 +1,328 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"io"
+
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	"github.com/lazyledger/nmt"
+
+	"github.com/ipfs/go-ipfs/core/coredag"
+	"github.com/ipfs/go-ipfs/plugin"
+	format "github.com/ipfs/go-ipld-format"
+	node "github.com/ipfs/go-ipld-format"
+	mh "github.com/multiformats/go-multihash"
+)
+
+const (
+	// NMT is the codec used for this plugin.
+	// 0x77 seems to be free:
+	// https://github.com/multiformats/multicodec/blob/master/table.csv
+	NMT = 0x77
```

Comment:
Do we want to settle on this?

Reply:
0x77 is fitting as it's LL upside down.

However I have doubts they will allow us to claim such a low value namespace for an application layer project, we might have to pick a higher one.

Reply:
> 0x77 is fitting as it's LL upside down.

That's why I liked it. You are probably right about the low value:

> The first 127 bits are encoded as a single-byte varint, hence they are reserved for the most widely used multicodecs. So if you are adding your own codec to the table, you most likely would want to ask for a codec bigger than 0x80.

Reply:
changed to `0x1077` for now